build-tools/packages/build-cli/src/test/library/release.test.ts

```diff
@@ -0,0 +1,113 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "chai";
+import { getInternalVersionRange } from "../../library/release.js";
+
+interface TestMatrix {
+	inputVersion: string;
+	subCases: {
+		[interval: number]: string;
+	};
+}
+
+describe("Legacy compatibility ranges", () => {
+	const testMatrix: TestMatrix[] = [
+		{
+			inputVersion: "2.0.9",
+			subCases: {
+				10: "2.10.0",
+				20: "2.20.0",
+				25: "2.25.0",
+				30: "2.30.0",
+				40: "2.40.0",
+				50: "2.50.0",
+				200: "2.200.0",
+			},
+		},
```

Comment:
I always find it really hard to read these. I often have to run the test suite to figure out how everything fits together. That is, you kind of need to learn how to "read" the test cases and translate them into expectations when you're reviewing them to know if they're correct. That's a me problem :) but I wonder if copy/pasting the printed output as a comment above the matrix might help folks like me.

To Jason's point on a separate thread, the test cases have great names, but since they're generated readers have to compile it in their heads (or run the tests), so sharing some sample output inline would be helpful.